src/field_player_avatar.c

```diff
@@ -1725,7 +1726,9 @@ static void Task_WaitStopSurfing(u8 taskId)
 
 #define FISHING_PROXIMITY_BOOST 4
 #define FISHING_STICKY_BOOST    36
-#define FISHING_DEFAULT_ODDS    50
+#define FISHING_OLD_ROD_ODDS    ((I_FISHING_BITE_ODDS >= GEN_4) ? 75 : ((I_FISHING_BITE_ODDS >= GEN_3) ? 50 : ((I_FISHING_BITE_ODDS >= GEN_1) ? 0 : 50)))
+#define FISHING_GOOD_ROD_ODDS   ((I_FISHING_BITE_ODDS >= GEN_4) ? 50 : ((I_FISHING_BITE_ODDS >= GEN_3) ? 50 : ((I_FISHING_BITE_ODDS >= GEN_1) ? 33 : 50)))
+#define FISHING_SUPER_ROD_ODDS  ((I_FISHING_BITE_ODDS >= GEN_4) ? 25 : ((I_FISHING_BITE_ODDS >= GEN_3) ? 50 : ((I_FISHING_BITE_ODDS >= GEN_1) ? 50 : 50)))
```

Comment:
This ternary stacking gets very confusing very quickly. I would put the defines inside preproc conditions for clarity